src/main/java/Duke.java

```diff
@@ -1,10 +1,333 @@
+import Exceptions.DukeException;
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Task;
+import Tasks.ToDo;
+import java.util.Scanner;
+import java.util.Arrays;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    /*Command Prefixes*/
+    private static final String COMMAND_LIST = "list";
+
+    /*Delimiter Prefixes*/
+    private static final String EVENT_QUALIFIER = "/at";
+    private static final String DEADLINE_QUALIFIER = "/by";
+
+    /*Constants used*/
+    private static final int MAX_TASKS = 100;
+
+    //private static boolean hasExited = false;
+    private static final Scanner input = new Scanner(System.in);
+
+    public static void main(String[] args) throws DukeException {
+        String userInput = "";
+        Task[] currentTasks = new Task[MAX_TASKS];
+        int numOfTasks = 0;
+        boolean isOngoing = true;
+
+        showWelcomeMessage();
+        executeCommands(currentTasks, numOfTasks, isOngoing);
+    }
+
+
+    /*
+     * ===========================================
+     *           MAIN COMMANDS
+     * ===========================================
+     */
+
+
+    /**
+     * Execute commands as intended by user
+     *
+     * @param currentTasks Tasks Array of all tasks
+     * @param numOfTasks   Number of tasks
+     * @param isOngoing    Whether the program has yet to be exited
+     * @return numOfTasks Updated number of tasks
+     * @throws DukeException to catch error
+     */
+    private static void executeCommands(Task[] currentTasks, int numOfTasks, boolean isOngoing) throws DukeException {
+        String userInput;
+        while (isOngoing) {
+            userInput = getUserInput();
+            String[] splitUserInput = splitCommands(userInput);
+            switch (splitUserInput[0].toLowerCase()) {
+            case "bye":
```

Comment:
Maybe you can try replacing string literals in each case statement into a constant. Avoid using string literals if possible :)

Reply:
Alright will do just that in my next commit! Thank you